internal/pkg/rpaas/k8s.go

```diff
@@ -708,7 +724,22 @@ func (m *k8sRpaasManager) GetInstanceAddress(ctx context.Context, name string) (
 
 func (m *k8sRpaasManager) GetInstance(ctx context.Context, name string) (*v1alpha1.RpaasInstance, error) {
 	var instance v1alpha1.RpaasInstance
-	err := m.cli.Get(ctx, types.NamespacedName{Name: name, Namespace: namespaceName()}, &instance)
+
+	poolNamespace, err := m.poolNamespace()
+	if err != nil {
+		return nil, err
+	}
+	if poolNamespace != "" {
+		err = m.cli.Get(ctx, types.NamespacedName{Name: name, Namespace: poolNamespace}, &instance)
+		if err != nil && !k8sErrors.IsNotFound(err) {
+			return nil, err
```

Comment:
This will give an error different from what we have now. And this is basically the same code as bellow. Maybe setting `namespaceName()` as the default namespace, and if `m.poolNamespace()` returns a string we just update its value ?

Reply:
Like what you did for `ensureNamespaceExists`

Reply:
@pedrokiefer  I cant change to that cause there is a previous-compatible to prevent to do 404 in a old resources.